openhcl/page_pool_alloc/src/lib.rs

```diff
@@ -47,14 +302,34 @@ enum State {
         device_id: usize,
         tag: String,
     },
+    AllocatedPendingRestore {
+        base_pfn: u64,
```

Comment:
Probably should lift the common fields here, too. I bet it simplifies your allocation loop.